src/emqx_mgmt_cli.erl

```diff
@@ -535,31 +535,37 @@ mnesia(_) ->
 %% @doc Trace Command
 
 trace(["list"]) ->
-    foreach(fun({{Who, Name}, LogFile}) ->
-                ?PRINT("trace ~s ~s -> ~s~n", [Who, Name, LogFile])
+    foreach(fun({{Who, Name}, {Level, LogFile}}) ->
+                ?PRINT("Trace(~s=~s, level=~s, destination=~p)~n", [Who, Name, Level, LogFile])
             end, emqx_tracer:lookup_traces());
 
-trace(["client", ClientId, "off"]) ->
+trace(["stop", "client", ClientId]) ->
     trace_off(client_id, ClientId);
 
-trace(["client", ClientId, LogFile]) ->
-    trace_on(client_id, ClientId, LogFile);
+trace(["start", "client", ClientId, LogFile]) ->
+    trace_on(client_id, ClientId, all, LogFile);
 
-trace(["topic", Topic, "off"]) ->
+trace(["start", "client", ClientId, Level, LogFile]) ->
+    trace_on(client_id, ClientId, Level, LogFile);
+
+trace(["stop", "topic", Topic]) ->
     trace_off(topic, Topic);
 
-trace(["topic", Topic, LogFile]) ->
-    trace_on(topic, Topic, LogFile);
+trace(["start", "topic", Topic, LogFile]) ->
+    trace_on(topic, Topic, all, LogFile);
+
+trace(["start", "topic", Topic, Level, LogFile]) ->
+    trace_on(topic, Topic, Level, LogFile);
 
 trace(_) ->
-    emqx_cli:usage([{"trace list",                       "List all traces"},
-                    {"trace client <ClientId> <LogFile>","Trace a client by client_id"},
-                    {"trace client <ClientId> off",      "Stop tracing a client by client_id"},
-                    {"trace topic <Topic> <LogFile>",    "Trace a topic"},
-                    {"trace topic <Topic> off",          "Stop tracing a Topic"}]).
-
-trace_on(Who, Name, LogFile) ->
-    case emqx_tracer:start_trace({Who, iolist_to_binary(Name)}, LogFile) of
+    emqx_cli:usage([{"trace list", "List all traces started"},
+                    {"trace start client <ClientId> [<Level>] <File>", "Trace logs related to a client_id"},
```

Comment:
I would reword to:
Traces for a client
Traces for a topic

Having the optional arg in the middle is a bit counterintuitive.
We will be in a very awkward corner when it’s time to add another optional arg

Reply:
fixed